spec/models/manageiq/providers/openstack/cloud_manager/refresh_worker_spec.rb

```diff
@@ -1,28 +1,11 @@
 describe ManageIQ::Providers::Openstack::CloudManager::RefreshWorker do
   context "EMS with children" do
-    let!(:network_manager) { FactoryBot.create(:ems_network) }
-    let!(:storage_manager) { FactoryBot.create(:ems_storage) }
-    let(:ems) do
-      FactoryBot.create(:ems_cloud).tap do |ems|
-        network_manager.update_attributes(:parent_ems_id => ems.id)
-        storage_manager.update_attributes(:parent_ems_id => ems.id)
-      end
-    end
+    let(:ems) { FactoryBot.create(:ems_openstack) }
 
     it ".queue_name_for_ems" do
       queue_name = described_class.queue_name_for_ems(ems)
-      expect(queue_name.count).to eq(3)
+      expect(queue_name.count).to eq(4)
```

Comment:
I'm looking for good reason why this should be 4 (Cloud+Network+Storage is 3)..

Reply:
Because we should be testing these the way we use them. Since this is now a leaf class with the correct inheritance set by the model, the counts all increase by one to reflect the correct inheritance. It makes no sense to use an intermediate class here since we would never actually do that in the product.